src/vl-progress-bar.js

```diff
@@ -49,6 +49,24 @@ export class VlProgressBar extends VlElement(HTMLElement) {
         this._progressBar.updateStep(this._shadow, number, focus);
     }
 
+    /**
+     * Geeft de progress bar DOM
+     * 
+     * @return {HTMLElement}
+     */
+    get element() {
+        return this._element;
+    }
+
+    /**
+     * Geeft van elke stap het button element.
+     * 
+     * @return {HTMLElement[]}
+     */
+    get buttons() {
+        return this._element.querySelectorAll(VlProgressBarStep.buttonSelector);
+    }
```

Comment:
niet duidelijker om alle stappen op te vragen en dan .map(stap => stap.button) te doen?

Reply:
De stappen hebben geen shadow DOM omdat dan de Webuniversum stijl breekt door slots. Dus is helaas niet mogelijk om het zo te doen.

Reply:
da snap ik niet wat dat met shadow dom te maken heeft?

Reply:
De (button) elementen zitten in de wizard, niet in de step.

Reply:
ah zo ne step is gelijk ne vl-rich-data-field? configuratie maar geen inhoud?

Reply:
Exact!

Reply:
maar dan zou ik dus de selector gewoon bij wizard steken en niet bij de step

Reply:
De wizard kent toch onmogelijk de juiste selecteer van de vl-ui-progress-bar afgeschermde DOM die in de toekomst ook nog eens kan wijzigen?

Reply:
excuses, bij de progress-bar, niet bij de progress-bar-step. als die zelf niks rendert weet die toch ook niks van selectors.

Reply:
De template (met Button declaratie) zit wel in de vl-progress-bar-step, dus lijkt me daar wel thuis te horen?

Reply:
Dan ben ik in de war met hierboven dat de step enkel configuratie zou zijn en geen inhoud ...

Reply:
Ja, ik zie het nu, er zit wel een "template" functie in die step. Ca va, je zou die ook kunnen verhuizen naar progress bar en dan step echt een lege doos laten zijn en dan moet je die selector niet exposen ... (maar dat gaat misschien te ver)

Reply:
Zo was het in #1 maar is op vraag van @SanderKleykens verplaatst, terecht lijkt me. Het is de step die de template bepaalt, de elementen worden toevallig (door omstandigheden) door de parent gerenderd.

Reply:
is goed

Reply:
methode kan dan dus protected nog gemaakt worden op de step? alleen de progress bar heeft die nodig en niet de afnemers?